editoast/src/modelsv2/work_schedules.rs

```diff
@@ -0,0 +1,53 @@
+use crate::schema::TrackRange;
+use chrono::NaiveDateTime;
+use editoast_derive::ModelV2;
+use strum_macros::FromRepr;
+
+use crate::error::Result;
+use crate::modelsv2::Changeset;
+use serde::Deserialize;
+use serde::Serialize;
+use utoipa::ToSchema;
+
+#[derive(Debug, Default, Clone, Copy, Serialize, Deserialize, FromRepr, ToSchema, PartialEq)]
+#[serde(rename_all = "UPPERCASE")]
+pub enum WorkScheduleType {
+    #[default]
+    Catenary,
+    Track,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, ModelV2, ToSchema)]
+#[model(table = crate::tables::work_schedule)]
+pub struct WorkSchedule {
+    pub id: i64,
+    pub start_date_time: NaiveDateTime,
+    pub end_date_time: NaiveDateTime,
+    #[model(json)]
+    pub track_ranges: Vec<TrackRange>,
+    pub obj_id: String,
+    #[model(to_enum)]
+    pub work_schedule_type: WorkScheduleType,
+}
+
+fn dates_in_order(a: Option<NaiveDateTime>, b: Option<NaiveDateTime>) -> bool {
+    match (a, b) {
+        (Some(a), Some(b)) => a <= b,
+        _ => true,
+    }
+}
+
+impl WorkSchedule {
+    pub fn validate(work_schedule_changeset: &Changeset<Self>) -> Result<()> {
+        if !dates_in_order(
+            work_schedule_changeset.start_date_time,
+            work_schedule_changeset.end_date_time,
+        ) {
+            return Err(
+                crate::views::work_schedules::WorkScheduleError::StartDateAfterEndDate.into(),
```

Comment:
A model cannot depend on a view. The standard way to validate a payload at deserialization is to implement the `Deserialization` trait manually. You can see examples here: https://github.com/osrd-project/osrd/blob/dev/editoast/src/schema/v2/trainschedule.rs#L45

Reply:
Fixed ! I implemented a custom Deserializer